### ISSUE TITLE
Question on the syntax of symbol declaration/definition

### ISSUE THREAD

**Issue by amelieled** [Deducteam/lambdapi#502]
```
opaque symbol eq_sym {a} (x y:τ a) : π (x=y) → π (y=x)
begin
  assume a x y xy
  symmetry
  apply xy
end
```
shouldn't be parsed?
Currently, the error is: `Unification goals are unsatisfiable.`.

**Comment by fblanqui** [comment]
Thanks to provide a standalone file.

**Comment by amelieled** [comment]
It's just an example. You can try by yourself with any files (just delete `:=`).

**Comment by fblanqui** [comment]
Bad answer. When you raise an issue, you should provide a standalone file so that we can reproduce the issue.

**Comment by amelieled** [comment]
When it's a specific one, yes, but it's not the case here.

**Comment by fblanqui** [comment]
It is. Why don't you try to be more helpful to identify what is actually the problem? It is common practice to ask for standalone and possibly minimal examples. See for instance https://github.com/coq/coq/issues/new .

**Comment by amelieled** [comment]
I know this "common practice" and the most important is that my explanation is self-sufficient. 
You can take the file tests/OK/natproofs.lp for example, but it's better if you take another example in order to, maybe, find a counter-example that proofs my generalization isn't correct.

**Comment by fblanqui** [comment]
The answer to your initial question is: no. As suggested by the error message: this is not a parsing problem. The fact that the declaration is opaque or not is not important. `symbol f : A begin ... end` is just a symbol declaration. The `begin ... end` part may be necessary to solve unification constraints: it is NOT a proof of `A`. For making a definition of `f` or providing a proof of `A`, which is the same thing, `:=` is mandatory before `begin ... end`.

**Comment by amelieled** [comment]
Ok, I see but to be honest, this is important and it's not in the documentation.

**Comment by fblanqui** [comment]
Fixed in #506
